abi/abi_encode.go

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2018 De-labtory
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package abi
+
+import (
+	"github.com/DE-labtory/koa/crpyto"
+	"github.com/DE-labtory/koa/encoding"
+)
+
+type Pointer []byte
+type Size []byte
+type Value []byte
+
+// Encode abi parameters
+func Encode(params ...interface{}) ([]byte, error) {
+	values, err := encodeValues(params...)
+	if err != nil {
+		return nil, err
+	}
+
+	sizes, err := encodeSizes(values)
+	if err != nil {
+		return nil, err
+	}
+
+	pointers, err := encodePointers(sizes, values)
+	if err != nil {
+		return nil, err
+	}
+
+	var Args []byte
+
+	for _, pointer := range pointers {
+		Args = append(Args, pointer...)
+	}
+
+	for index := 0; index < len(params); index++ {
+		Args = append(Args, sizes[index]...)
+		Args = append(Args, values[index]...)
+	}
+
+	return Args, nil
+}
+
+func encodeValues(params ...interface{}) ([]Value, error) {
+	values := make([]Value, len(params))
+
+	for index, param := range params {
+		bytesValue, err := encoding.EncodeOperand(param, 0)
```

Comment:
we can use `NO_PADDING` const instead of just 0

Reply:
fixed it!